# Life Claims Intelligence — Claims Reliability Index (CRI)

## Objective

The Claims Reliability Index (CRI) is an independent analytical framework
developed in this project to evaluate insurer reliability beyond the
headline Claim Settlement Ratio (CSR).

The index combines multiple claims-reliability dimensions into a common
score so that insurers can be compared using a broader view of claim
settlement performance.

> **Important:** CRI is an independent analytical index developed for this
> project. It is not an IRDAI-official score or regulatory rating.

## Why CRI?

A high count-based Claim Settlement Ratio does not necessarily imply that
an insurer settles a similarly high proportion of claim value.

Therefore, the CRI considers value-based settlement performance,
count–value consistency, and claim outcome/processing reliability rather
than relying on a single headline metric.

## Life-specific Adaptation

The original project framework proposed four CRI components. During actual
Life Insurance data analysis, Complaint Rate and Claim Value Disparity could
not be retained as reliable Life-level components.

Therefore, the Life CRI will be adapted to the metrics that are actually
available and validated in the Life dataset.

The methodology will be documented transparently rather than introducing
unsupported or fabricated variables.

----

## Load Dataset

In [1]:
# importing libraries

import pandas as pd 
import numpy as np 

In [2]:
df = pd.read_csv(r"D:\My Projects\Insurance_Claims_Intelligence\data\clean\Life_Claims_Feature_Engineered.csv")

df

,FY,Insurer,pending_start_count,pending_start_amount,intimated_count,intimated_amount,total_claims_count,total_claims_amount,paid_count,paid_amount,...,pending_6m_to_1y,pending_gt_1y,Low_Volume_Flag,Count_CSR,Value_CSR,Count_Value_Gap,Unclaimed_Ratio,Pending_Rate,Long_Pending_Share,Repudiation_Rate
0,2020-21,LIC,5875,349.690000,941101,18755.650000,946976,19105.340000,933889,18295.580000,...,0.0,0.0,False,98.618022,95.761604,2.856418,1.237821,0.182159,0.000000,0.689669
1,2020-21,Aditya Birla Sun Life,19,3.802883,6455,468.846357,6474,472.649240,6347,440.264288,...,0.0,0.0,False,98.038307,93.148206,4.890101,0.000000,0.169910,0.000000,1.791783
2,2020-21,Bandhan,0,0.000000,401,107.440000,401,107.440000,398,105.980000,...,0.0,0.0,True,99.251870,98.641102,0.610768,0.000000,0.000000,NaN,0.748130
3,2020-21,Ageas Federal,5,1.255000,1800,87.051683,1805,88.306683,1716,73.848647,...,0.0,0.0,False,95.069252,83.627473,11.441779,0.000000,2.770083,0.000000,2.105263
4,2020-21,Aviva,5,0.778359,1050,116.371251,1055,117.149610,1034,111.572118,...,0.0,0.0,False,98.009479,95.239000,2.770478,0.000000,0.000000,NaN,1.990521
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
117,2024-25,Sahara,0,0.000000,0,0.000000,0,0.000000,0,0.000000,...,0.0,0.0,True,NaN,NaN,NaN,NaN,NaN,NaN,NaN
118,2024-25,SBI Life,118,23.580493,44831,2598.751947,44949,2622.332439,44205,2497.939508,...,34.0,2.0,False,98.344791,95.256401,3.088390,0.290204,0.367083,1.212121,1.165766
119,2024-25,Shriram,6,0.294771,4846,180.195625,4852,180.490395,4770,147.476986,...,0.0,0.0,False,98.309975,81.709049,16.600926,0.000000,0.144270,0.000000,1.401484
120,2024-25,Star Union,3,0.805000,2410,144.384758,2413,145.189758,2385,140.738808,...,0.0,0.0,False,98.839619,96.934391,1.905228,0.000000,0.082884,0.000000,1.077497


----

## 1. CRI Component Design

The Life CRI will be constructed using only metrics that are available,
reliable, and analytically meaningful in the final Life Insurance dataset.

### Candidate Metrics

| Metric | Reliability Direction | Initial Role |
|---|---|---|
| Value_CSR | Higher is better | Core |
| Count_Value_Gap | Lower is better | Core |
| Repudiation_Rate | Lower is better | Candidate |
| Pending_Rate | Lower is better | Candidate |
| Unclaimed_Ratio | Lower is better | Candidate |
| Long_Pending_Share | Lower is better | Supporting |
| Count_CSR | Higher is better | Supporting |

The final CRI components will be selected based on business meaning,
redundancy, and availability of reliable data.

Metrics that are useful for interpretation but would create excessive
overlap or redundancy will remain as supporting diagnostic metrics rather
than receiving an independent CRI weight.

----

## Final Life CRI Components

Based on the structure and availability of the Life Insurance dataset,
the CRI will use three primary components:

1. **Value Reliability**
   - Value_CSR
   - Higher values indicate better value-based claim settlement.

2. **Count–Value Consistency**
   - Count_Value_Gap
   - Lower values indicate greater consistency between count-based and
     value-based settlement performance.

3. **Outcome Reliability**
   - Repudiation_Rate
   - Pending_Rate
   - Unclaimed_Ratio
   - These metrics will be combined into a single outcome-reliability
     component because they represent related claim-outcome dimensions.

### Supporting Metrics

Count_CSR and Long_Pending_Share will be retained as supporting
diagnostic metrics. They will not receive independent CRI weights in order
to reduce redundancy and avoid giving excessive weight to closely related
claim-status measures.

### Baseline Weighting

The three primary CRI components will initially receive equal weights:

- Value Reliability: 33.33%
- Count–Value Consistency: 33.33%
- Outcome Reliability: 33.33%

These baseline weights will later be subjected to sensitivity analysis.

----

### Inspecting 

In [3]:
df[[
    "Value_CSR",
    "Count_Value_Gap",
    "Repudiation_Rate",
    "Pending_Rate",
    "Unclaimed_Ratio",
    "Long_Pending_Share",
    "Count_CSR"
]].describe()

,Value_CSR,Count_Value_Gap,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio,Long_Pending_Share,Count_CSR
count,116.000000,116.000000,116.000000,116.000000,116.000000,84.000000,116.000000
mean,93.212092,4.153878,2.039074,0.487680,0.077641,9.226444,97.365970
std,9.924512,4.046813,6.136775,3.104228,0.245137,22.426659,9.202453
min,0.000000,-0.297851,0.096167,0.000000,0.000000,0.000000,0.000000
25%,92.744876,1.689739,0.799660,0.000000,0.000000,0.000000,97.856519
50%,95.324729,2.999698,1.194237,0.067987,0.000000,0.000000,98.518600
75%,96.976318,5.516924,1.865060,0.183848,0.000000,0.000000,99.093611
max,99.798183,19.773667,66.666667,33.333333,1.560407,100.000000,99.726027


### Checking for null values

In [4]:
df[[
    "Value_CSR",
    "Count_Value_Gap",
    "Repudiation_Rate",
    "Pending_Rate",
    "Unclaimed_Ratio",
    "Long_Pending_Share",
    "Count_CSR"
]].isna().sum()

Value_CSR              6
Count_Value_Gap        6
Repudiation_Rate       6
Pending_Rate           6
Unclaimed_Ratio        6
Long_Pending_Share    38
Count_CSR              6
dtype: int64

In [5]:
df[df["Value_CSR"].isna()][[
    "Insurer",
    "FY",
    "Value_CSR",
    "Count_Value_Gap",
    "Repudiation_Rate",
    "Pending_Rate",
    "Unclaimed_Ratio",
    "Long_Pending_Share",
    "Count_CSR"
]]

,Insurer,FY,Value_CSR,Count_Value_Gap,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio,Long_Pending_Share,Count_CSR
72,Acko Life,2023-24,NaN,NaN,NaN,NaN,NaN,NaN,NaN
79,Credit Access Life,2023-24,NaN,NaN,NaN,NaN,NaN,NaN,NaN
83,Godigit Life,2023-24,NaN,NaN,NaN,NaN,NaN,NaN,NaN
97,Acko Life,2024-25,NaN,NaN,NaN,NaN,NaN,NaN,NaN
104,Credit Access Life,2024-25,NaN,NaN,NaN,NaN,NaN,NaN,NaN
117,Sahara,2024-25,NaN,NaN,NaN,NaN,NaN,NaN,NaN


----

## CRI Data Availability Check

During the EDA and feature-engineering stages, six insurer-year observations
were identified where the underlying claims data required to calculate the
engineered reliability metrics was unavailable.

The affected observations are:

- Acko Life — 2023-24
- Credit Access Life — 2023-24
- Godigit Life — 2023-24
- Acko Life — 2024-25
- Credit Access Life — 2024-25
- Sahara — 2024-25

These observations contain missing values across the core CRI metrics.

The missing values are treated as unavailable data rather than zero
performance. Therefore, these observations will be excluded from the CRI
calculation but retained in the master dataset for transparency and
auditability.

`Long_Pending_Share` has additional missing values. Since it is a supporting
diagnostic metric rather than a primary CRI component, its missingness does
not affect the core CRI calculation.

---

### Creating working copy of original dataset 

In [6]:
required_metrics = [
    "Value_CSR",
    "Count_Value_Gap",
    "Repudiation_Rate",
    "Pending_Rate",
    "Unclaimed_Ratio"
]

cri_df = df[
    ["Insurer", "FY"] + required_metrics
].dropna(
    subset=required_metrics
).copy()

cri_df.shape

(116, 7)

In [7]:
cri_df[required_metrics].isna().sum()

Value_CSR           0
Count_Value_Gap     0
Repudiation_Rate    0
Pending_Rate        0
Unclaimed_Ratio     0
dtype: int64

In [8]:
cri_df.head()

,Insurer,FY,Value_CSR,Count_Value_Gap,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio
0,LIC,2020-21,95.761604,2.856418,0.689669,0.182159,1.237821
1,Aditya Birla Sun Life,2020-21,93.148206,4.890101,1.791783,0.169910,0.000000
2,Bandhan,2020-21,98.641102,0.610768,0.748130,0.000000,0.000000
3,Ageas Federal,2020-21,83.627473,11.441779,2.105263,2.770083,0.000000
4,Aviva,2020-21,95.239000,2.770478,1.990521,0.000000,0.000000


----

### 1. Creating Value_Reliability_Score

In [9]:
cri_df["Value_Reliability_Score"] = (
    (cri_df["Value_CSR"] - cri_df["Value_CSR"].min())
    /
    (cri_df["Value_CSR"].max() - cri_df["Value_CSR"].min())
)

In [10]:
cri_df[
    ["Value_CSR", "Value_Reliability_Score"]
].head(10)

,Value_CSR,Value_Reliability_Score
0,95.761604,0.959553
1,93.148206,0.933366
2,98.641102,0.988406
3,83.627473,0.837966
4,95.239000,0.954316
5,91.969401,0.921554
6,98.547260,0.987465
7,92.605701,0.927930
8,87.980492,0.881584
9,93.282098,0.934707


In [11]:
cri_df["Value_Reliability_Score"].describe()

count    116.000000
mean       0.934006
std        0.099446
min        0.000000
25%        0.929324
50%        0.955175
75%        0.971724
max        1.000000
Name: Value_Reliability_Score, dtype: float64

---

### 2.Creating Count_Value_Consistency_Score

In [12]:
cri_df["Absolute_Count_Value_Gap"] = cri_df["Count_Value_Gap"].abs()

In [13]:
cri_df["Count_Value_Consistency_Score"] = (
    (
        cri_df["Absolute_Count_Value_Gap"].max()
        - cri_df["Absolute_Count_Value_Gap"]
    )
    /
    (
        cri_df["Absolute_Count_Value_Gap"].max()
        - cri_df["Absolute_Count_Value_Gap"].min()
    )
)


In [14]:
cri_df[
    [
        "Count_Value_Gap",
        "Absolute_Count_Value_Gap",
        "Count_Value_Consistency_Score"
    ]
].head(10)

,Count_Value_Gap,Absolute_Count_Value_Gap,Count_Value_Consistency_Score
0,2.856418,2.856418,0.855544
1,4.890101,4.890101,0.752696
2,0.610768,0.610768,0.969112
3,11.441779,11.441779,0.421363
4,2.770478,2.770478,0.859891
5,6.509633,6.509633,0.670793
6,0.501868,0.501868,0.974619
7,4.498038,4.498038,0.772524
8,9.031461,9.031461,0.543258
9,5.253135,5.253135,0.734337


----

### 3. Pending Rate

In [15]:
cri_df["Pending_Score"] = (
    (cri_df["Pending_Rate"].max() - cri_df["Pending_Rate"])
    /
    (cri_df["Pending_Rate"].max() - cri_df["Pending_Rate"].min())
)

### 4. Unclaimed Ratio

In [16]:
cri_df["Unclaimed_Score"] = (
    (cri_df["Unclaimed_Ratio"].max() - cri_df["Unclaimed_Ratio"])
    /
    (cri_df["Unclaimed_Ratio"].max() - cri_df["Unclaimed_Ratio"].min())
)

### 5. Repudiation Rate

In [17]:
cri_df["Repudiation_Score"] = (
    (cri_df["Repudiation_Rate"].max() - cri_df["Repudiation_Rate"])
    /
    (cri_df["Repudiation_Rate"].max() - cri_df["Repudiation_Rate"].min())
)

---

### Outcome Reliability

In [18]:
cri_df["Outcome_Reliability_Score"] = (
    cri_df[
        [
            "Repudiation_Score",
            "Pending_Score",
            "Unclaimed_Score"
        ]
    ].mean(axis=1)
)

### Descriptive summary for all 

In [19]:
cri_df[
    [
        "Value_Reliability_Score",
        "Count_Value_Consistency_Score",
        "Repudiation_Score",
        "Pending_Score",
        "Unclaimed_Score",
        "Outcome_Reliability_Score"
    ]
].describe()

,Value_Reliability_Score,Count_Value_Consistency_Score,Repudiation_Score,Pending_Score,Unclaimed_Score,Outcome_Reliability_Score
count,116.000000,116.000000,116.000000,116.000000,116.000000,116.000000
mean,0.934006,0.789330,0.970814,0.985370,0.950243,0.968809
std,0.099446,0.204035,0.092185,0.093127,0.157098,0.080395
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.333333
25%,0.929324,0.720996,0.973428,0.994485,1.000000,0.981846
50%,0.955175,0.848298,0.983505,0.997960,1.000000,0.991072
75%,0.971724,0.914546,0.989432,1.000000,1.000000,0.995564
max,1.000000,1.000000,1.000000,1.000000,1.000000,0.999110


-----

## CRI

In [20]:
cri_df["CRI"] = (
    cri_df[
        [
            "Value_Reliability_Score",
            "Count_Value_Consistency_Score",
            "Outcome_Reliability_Score"
        ]
    ].mean(axis=1) * 100
)

In [21]:
cri_df[
    [
        "Insurer",
        "FY",
        "Value_Reliability_Score",
        "Count_Value_Consistency_Score",
        "Outcome_Reliability_Score",
        "CRI"
    ]
].head(10)

,Insurer,FY,Value_Reliability_Score,Count_Value_Consistency_Score,Outcome_Reliability_Score,CRI
0,LIC,2020-21,0.959553,0.855544,0.730784,84.862689
1,Aditya Birla Sun Life,2020-21,0.933366,0.752696,0.989811,89.195754
2,Bandhan,2020-21,0.988406,0.969112,0.996735,98.475110
3,Ageas Federal,2020-21,0.837966,0.421363,0.962239,74.052262
4,Aviva,2020-21,0.954316,0.859891,0.990515,93.490701
5,Bajaj Allianz,2020-21,0.921554,0.670793,0.992692,86.167940
6,Bharti Axa,2020-21,0.987465,0.974619,0.995720,98.593505
7,Canara HSBC,2020-21,0.927930,0.772524,0.979406,89.328666
8,Edelweiss Life,2020-21,0.881584,0.543258,0.983531,80.279094
9,Exide Life,2020-21,0.934707,0.734337,0.986921,88.532172


In [22]:
cri_df["CRI"].describe()

count    116.000000
mean      89.738153
std        9.596755
min       44.444444
25%       87.077442
50%       91.988899
75%       95.762996
max       99.848687
Name: CRI, dtype: float64

In [23]:
cri_df[
    ["Insurer", "FY", "CRI"]
].sort_values("CRI", ascending=False).head(10)

,Insurer,FY,CRI
99,Bandhan,2024-25,99.848687
50,Bandhan,2022-23,99.763328
52,Aviva,2022-23,99.433671
114,PNB Met Life,2024-25,99.322396
28,Aviva,2021-22,99.211985
76,Aviva,2023-24,99.210378
105,Canara HSBC,2024-25,99.094634
26,Bandhan,2021-22,99.023991
43,Sahara,2021-22,98.789887
6,Bharti Axa,2020-21,98.593505


----

### clean baseline column

In [24]:
cri_df["CRI_Baseline"] = cri_df["CRI"]

### Creating sensitivity scores:

In [25]:
cri_df["CRI_Value_Focused"] = (
    cri_df["Value_Reliability_Score"] * 0.40
    + cri_df["Count_Value_Consistency_Score"] * 0.30
    + cri_df["Outcome_Reliability_Score"] * 0.30
) * 100

In [26]:
cri_df["CRI_Consistency_Focused"] = (
    cri_df["Value_Reliability_Score"] * 0.30
    + cri_df["Count_Value_Consistency_Score"] * 0.40
    + cri_df["Outcome_Reliability_Score"] * 0.30
) * 100

In [27]:
cri_df["CRI_Outcome_Focused"] = (
    cri_df["Value_Reliability_Score"] * 0.30
    + cri_df["Count_Value_Consistency_Score"] * 0.30
    + cri_df["Outcome_Reliability_Score"] * 0.40
) * 100

In [28]:
cri_df[
    [
        "CRI_Baseline",
        "CRI_Value_Focused",
        "CRI_Consistency_Focused",
        "CRI_Outcome_Focused"
    ]
].describe()

,CRI_Baseline,CRI_Value_Focused,CRI_Consistency_Focused,CRI_Outcome_Focused
count,116.000000,116.000000,116.000000,116.000000
mean,89.738153,90.104397,88.657634,90.452428
std,9.596755,9.455303,10.387531,9.056760
min,44.444444,40.000000,50.000000,43.333333
25%,87.077442,87.676966,85.416563,88.324185
50%,91.988899,92.365419,90.992414,92.764817
75%,95.762996,95.901627,95.244906,96.125489
max,99.848687,99.863819,99.827328,99.854916


----

### Ranking Stability

In [29]:
cri_df["Rank_Baseline"] = cri_df["CRI_Baseline"].rank(
    ascending=False,
    method="min"
)

cri_df["Rank_Value_Focused"] = cri_df["CRI_Value_Focused"].rank(
    ascending=False,
    method="min"
)

cri_df["Rank_Consistency_Focused"] = cri_df["CRI_Consistency_Focused"].rank(
    ascending=False,
    method="min"
)

cri_df["Rank_Outcome_Focused"] = cri_df["CRI_Outcome_Focused"].rank(
    ascending=False,
    method="min"
)

In [30]:
cri_df[
    [
        "Insurer",
        "FY",
        "Rank_Baseline",
        "Rank_Value_Focused",
        "Rank_Consistency_Focused",
        "Rank_Outcome_Focused"
    ]
].head(10)

,Insurer,FY,Rank_Baseline,Rank_Value_Focused,Rank_Consistency_Focused,Rank_Outcome_Focused
0,LIC,2020-21,99.0,97.0,93.0,100.0
1,Aditya Birla Sun Life,2020-21,77.0,77.0,77.0,75.0
2,Bandhan,2020-21,11.0,11.0,13.0,11.0
3,Ageas Federal,2020-21,108.0,108.0,108.0,107.0
4,Aviva,2020-21,50.0,50.0,50.0,49.0
5,Bajaj Allianz,2020-21,94.0,94.0,97.0,94.0
6,Bharti Axa,2020-21,10.0,10.0,10.0,10.0
7,Canara HSBC,2020-21,76.0,76.0,76.0,74.0
8,Edelweiss Life,2020-21,103.0,103.0,103.0,102.0
9,Exide Life,2020-21,81.0,81.0,81.0,80.0


### Actual stability number


In [31]:
rank_cols = [
    "Rank_Baseline",
    "Rank_Value_Focused",
    "Rank_Consistency_Focused",
    "Rank_Outcome_Focused"
]

cri_df[rank_cols].corr(method="spearman")

,Rank_Baseline,Rank_Value_Focused,Rank_Consistency_Focused,Rank_Outcome_Focused
Rank_Baseline,1.000000,0.999769,0.998854,0.999116
Rank_Value_Focused,0.999769,1.000000,0.998670,0.998962
Rank_Consistency_Focused,0.998854,0.998670,1.000000,0.996817
Rank_Outcome_Focused,0.999116,0.998962,0.996817,1.000000


----

## Final sanity

### Top 5 CRI

In [32]:
top_cri = cri_df[
    [
        "Insurer",
        "FY",
        "CRI",
        "Value_Reliability_Score",
        "Count_Value_Consistency_Score",
        "Outcome_Reliability_Score",
        "Repudiation_Rate",
        "Pending_Rate",
        "Unclaimed_Ratio"
    ]
].sort_values(
    "CRI",
    ascending=False
).head(5)

top_cri

,Insurer,FY,CRI,Value_Reliability_Score,Count_Value_Consistency_Score,Outcome_Reliability_Score,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio
99,Bandhan,2024-25,99.848687,1.000000,0.996351,0.999110,0.273973,0.0,0.0
50,Bandhan,2022-23,99.763328,0.995665,0.999922,0.997312,0.632911,0.0,0.0
52,Aviva,2022-23,99.433671,0.989415,0.999349,0.994246,1.245330,0.0,0.0
114,PNB Met Life,2024-25,99.322396,0.995040,0.986282,0.998350,0.425607,0.0,0.0
28,Aviva,2021-22,99.211985,0.986305,0.997659,0.992396,1.614833,0.0,0.0


### Bottom 5 CRI

In [33]:
bottom_cri = cri_df[
    [
        "Insurer",
        "FY",
        "CRI",
        "Value_Reliability_Score",
        "Count_Value_Consistency_Score",
        "Outcome_Reliability_Score",
        "Repudiation_Rate",
        "Pending_Rate",
        "Unclaimed_Ratio"
    ]
].sort_values(
    "CRI",
    ascending=True
).head(5)

bottom_cri

,Insurer,FY,CRI,Value_Reliability_Score,Count_Value_Consistency_Score,Outcome_Reliability_Score,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio
108,Godigit Life,2024-25,44.444444,0.000000,1.000000,0.333333,66.666667,33.333333,0.000000
68,Shriram,2022-23,58.877589,0.777853,0.000000,0.988475,2.148389,0.124906,0.000000
11,HDFC Life,2020-21,61.005259,0.802237,0.092070,0.935851,0.494816,1.048539,0.241867
21,Shriram,2020-21,62.102689,0.776326,0.107868,0.978887,3.716766,0.298426,0.000000
93,Shriram,2023-24,63.020320,0.803323,0.093785,0.993502,1.084431,0.154919,0.000000


---

## CRI Methodology — Final

The Life Claims Reliability Index (CRI) is an independent analytical index
developed in this project to summarize insurer reliability across multiple
claims-performance dimensions.

It is not an IRDAI-official score or regulatory rating.

### CRI Components

The Life-specific CRI uses three primary components:

1. **Value Reliability**
   - Based on Value_CSR.
   - Higher Value_CSR indicates better value-based claim settlement
     performance.

2. **Count–Value Consistency**
   - Based on the absolute Count–Value Gap.
   - The gap is defined as Count_CSR − Value_CSR.
   - Because consistency is about the distance between the two measures,
     the absolute gap is used.
   - Smaller absolute gaps receive higher consistency scores.

3. **Outcome Reliability**
   - Combines Repudiation_Rate, Pending_Rate, and Unclaimed_Ratio.
   - Each metric is normalized separately and directionally aligned so
     that higher normalized scores indicate better reliability.
   - The three normalized scores are averaged to create the
     Outcome_Reliability_Score.

### Normalization

Min-max normalization was used to place metrics on a common 0–1 scale.

For higher-is-better metrics:

    (x − min(x)) / (max(x) − min(x))

For lower-is-better metrics:

    (max(x) − x) / (max(x) − min(x))

### Baseline Weighting

The three CRI components receive equal baseline weights:

- Value Reliability: 33.33%
- Count–Value Consistency: 33.33%
- Outcome Reliability: 33.33%

The final CRI is calculated on a 0–100 scale.

### Data Availability

Six insurer-year observations did not contain the core metrics required for
CRI calculation. These observations were excluded from the CRI working
dataset but retained in the master dataset.

### Sensitivity Analysis

The baseline weighting was tested against three alternative scenarios:

- Value-focused: 40% / 30% / 30%
- Consistency-focused: 30% / 40% / 30%
- Outcome-focused: 30% / 30% / 40%

Spearman correlations between the baseline and alternative rankings were
approximately 0.999, indicating that the insurer-year ranking was highly
stable under these moderate weight changes.

### Sanity Check

High-CRI observations generally showed strong performance across the three
CRI components, while low-CRI observations showed weakness in one or more
major components.

This confirms that the index behaves directionally as intended. The sanity
check is not treated as independent statistical validation.

----

## Final CRI table

In [34]:
final_cri = cri_df[
    [
        "Insurer",
        "FY",
        "Value_CSR",
        "Count_Value_Gap",
        "Repudiation_Rate",
        "Pending_Rate",
        "Unclaimed_Ratio",
        "Value_Reliability_Score",
        "Count_Value_Consistency_Score",
        "Outcome_Reliability_Score",
        "CRI"
    ]
].copy()

final_cri.head()

,Insurer,FY,Value_CSR,Count_Value_Gap,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio,Value_Reliability_Score,Count_Value_Consistency_Score,Outcome_Reliability_Score,CRI
0,LIC,2020-21,95.761604,2.856418,0.689669,0.182159,1.237821,0.959553,0.855544,0.730784,84.862689
1,Aditya Birla Sun Life,2020-21,93.148206,4.890101,1.791783,0.169910,0.000000,0.933366,0.752696,0.989811,89.195754
2,Bandhan,2020-21,98.641102,0.610768,0.748130,0.000000,0.000000,0.988406,0.969112,0.996735,98.475110
3,Ageas Federal,2020-21,83.627473,11.441779,2.105263,2.770083,0.000000,0.837966,0.421363,0.962239,74.052262
4,Aviva,2020-21,95.239000,2.770478,1.990521,0.000000,0.000000,0.954316,0.859891,0.990515,93.490701


## CRI ranking

In [35]:
final_cri = final_cri.sort_values(
    ["FY", "CRI"],
    ascending=[True, False]
).reset_index(drop=True)

final_cri.head(10)

,Insurer,FY,Value_CSR,Count_Value_Gap,Repudiation_Rate,Pending_Rate,Unclaimed_Ratio,Value_Reliability_Score,Count_Value_Consistency_Score,Outcome_Reliability_Score,CRI
0,Bharti Axa,2020-21,98.547260,0.501868,0.950872,0.000000,0.0,0.987465,0.974619,0.995720,98.593505
1,Bandhan,2020-21,98.641102,0.610768,0.748130,0.000000,0.0,0.988406,0.969112,0.996735,98.475110
2,Sahara,2020-21,97.472992,-0.297851,1.807910,0.225989,0.0,0.976701,0.984937,0.989169,98.360237
3,Pramerica Life,2020-21,97.009785,1.599180,1.236476,0.154560,0.0,0.972060,0.919126,0.992745,96.131002
4,Aviva,2020-21,95.239000,2.770478,1.990521,0.000000,0.0,0.954316,0.859891,0.990515,93.490701
5,Kotak Mahindra,2020-21,95.471236,3.029445,1.135847,0.363471,0.0,0.956643,0.846794,0.991159,93.153212
6,Reliance Nippon,2020-21,95.031423,3.460506,1.465590,0.042481,0.0,0.952236,0.824994,0.992718,92.331614
7,Axis Max Life,2020-21,95.419093,3.932593,0.643327,0.004987,0.0,0.956121,0.801120,0.997210,91.815021
8,Star Union,2020-21,92.791267,3.172035,3.547401,0.489297,0.0,0.929789,0.839583,0.977826,91.573264
9,PNB Met Life,2020-21,93.924878,4.250098,1.825024,0.000000,0.0,0.941148,0.785063,0.991343,90.585138


### final sanity check

In [37]:
print("Master rows:", df.shape[0])
print("Master unique insurers:", df["Insurer"].nunique())

print("CRI rows:", final_cri.shape[0])
print("CRI unique insurers:", final_cri["Insurer"].nunique())
print("CRI unique FYs:", final_cri["FY"].nunique())
print("Missing CRI:", final_cri["CRI"].isna().sum())

print(
    "Insurers not represented in CRI:",
    sorted(set(df["Insurer"]) - set(final_cri["Insurer"]))
)

Master rows: 122
Master unique insurers: 27
CRI rows: 116
CRI unique insurers: 25
CRI unique FYs: 5
Missing CRI: 0
Insurers not represented in CRI: ['Acko Life', 'Credit Access Life']


----

# CRI — Conclusion

The Life Claims Reliability Index (CRI) was developed to provide a consolidated
view of insurer-level claims reliability across multiple performance dimensions.

The final CRI combines three components:

- **Value Reliability** — based on Value_CSR
- **Count–Value Consistency** — based on the absolute Count–Value Gap
- **Outcome Reliability** — based on Repudiation Rate, Pending Rate, and
  Unclaimed Ratio

All component metrics were normalized to a common 0–1 scale and combined using
equal baseline weights. The final CRI is reported on a 0–100 scale.

A total of **116 insurer-year observations** were available for CRI calculation.
Six insurer-year observations were excluded from the CRI working dataset because
the core metrics required for CRI calculation were unavailable. These
observations remain retained in the master dataset.

### Sensitivity Analysis

The baseline equal-weight CRI was tested against alternative weighting
scenarios. The resulting ranking correlations were approximately **0.999**
across the tested scenarios, indicating that the insurer-year ranking was highly
stable under moderate changes in component weights.

### Sanity Check

The highest-CRI observations generally showed strong performance across the
three CRI components, while lower-CRI observations showed weakness in one or
more components. This indicates that the CRI behaves directionally as intended.

The CRI is an **independent analytical index developed for this project** and
should not be interpreted as an official regulatory or IRDAI rating.

----

# End of CRI Phase

The Life Claims Reliability Index has now been constructed, validated through
sensitivity analysis, and reviewed through a component-level sanity check.

The resulting CRI dataset will serve as an analytical input for the next stages
of the project.

### Next Phase

**SQL-Based Claims Intelligence**

The next phase will use the finalized Life claims and CRI data to answer
business-oriented questions involving:

- Year-wise insurer ranking
- CRI movement over time
- Insurer improvement and deterioration
- Persistent reliability patterns
- Claims volume versus reliability
- Component-level performance analysis
- Advanced SQL analysis using CTEs and window functions

**Status: CRI Phase — COMPLETE**

----

### Thank you !